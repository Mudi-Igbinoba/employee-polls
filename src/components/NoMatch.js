import React from 'react';
import { Button } from 'react-bootstrap';
import { FcHighPriority } from 'react-icons/fc';
import { Link } from 'react-router-dom';

const NoMatch = () => {
  return (
    <section className='tw-text-center tw-h-screen tw-py-20 tw-space-y-9'>
      <FcHighPriority className='mx-auto tw-text-9xl' />

      <h1>404!</h1>

      <p>
        Sorry, it seems like you lost your way. Click this link and you'll be
        fine.
      </p>

      <Button as={Link} to='/login' variant='primary'>
        Go to Login
      </Button>
    </section>
  );
};

export default NoMatch;
