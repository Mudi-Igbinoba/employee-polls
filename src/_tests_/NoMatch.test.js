import React from 'react';
import { render } from '@testing-library/react';
import NoMatch from '../components/NoMatch';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import store from '../store';

describe('No Match', () => {
  it('will match snapshot', async () => {
    var view = render(
      <Provider store={store}>
        <MemoryRouter>
          <NoMatch />
        </MemoryRouter>
      </Provider>
    );
    expect(view).toMatchSnapshot();
  });
});
