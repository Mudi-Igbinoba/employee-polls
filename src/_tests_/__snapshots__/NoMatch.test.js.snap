// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`No Match will match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="tw-text-center tw-h-screen tw-py-20 tw-space-y-9"
      >
        <svg
          class="mx-auto tw-text-9xl"
          enable-background="new 0 0 48 48"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          version="1"
          viewBox="0 0 48 48"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M21.2,44.8l-18-18c-1.6-1.6-1.6-4.1,0-5.7l18-18c1.6-1.6,4.1-1.6,5.7,0l18,18c1.6,1.6,1.6,4.1,0,5.7l-18,18 C25.3,46.4,22.7,46.4,21.2,44.8z"
            fill="#F44336"
          />
          <path
            d="M21.6,32.7c0-0.3,0.1-0.6,0.2-0.9c0.1-0.3,0.3-0.5,0.5-0.7c0.2-0.2,0.5-0.4,0.8-0.5s0.6-0.2,1-0.2 s0.7,0.1,1,0.2c0.3,0.1,0.6,0.3,0.8,0.5c0.2,0.2,0.4,0.4,0.5,0.7c0.1,0.3,0.2,0.6,0.2,0.9s-0.1,0.6-0.2,0.9s-0.3,0.5-0.5,0.7 c-0.2,0.2-0.5,0.4-0.8,0.5c-0.3,0.1-0.6,0.2-1,0.2s-0.7-0.1-1-0.2s-0.5-0.3-0.8-0.5c-0.2-0.2-0.4-0.4-0.5-0.7S21.6,33.1,21.6,32.7z M25.8,28.1h-3.6L21.7,13h4.6L25.8,28.1z"
            fill="#fff"
          />
        </svg>
        <h1>
          404!
        </h1>
        <p>
          Sorry, it seems like you lost your way. Click this link and you'll be fine.
        </p>
        <a
          class="btn btn-primary"
          href="/login"
          role="button"
          tabindex="0"
        >
          Go to Login
        </a>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="tw-text-center tw-h-screen tw-py-20 tw-space-y-9"
    >
      <svg
        class="mx-auto tw-text-9xl"
        enable-background="new 0 0 48 48"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        version="1"
        viewBox="0 0 48 48"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M21.2,44.8l-18-18c-1.6-1.6-1.6-4.1,0-5.7l18-18c1.6-1.6,4.1-1.6,5.7,0l18,18c1.6,1.6,1.6,4.1,0,5.7l-18,18 C25.3,46.4,22.7,46.4,21.2,44.8z"
          fill="#F44336"
        />
        <path
          d="M21.6,32.7c0-0.3,0.1-0.6,0.2-0.9c0.1-0.3,0.3-0.5,0.5-0.7c0.2-0.2,0.5-0.4,0.8-0.5s0.6-0.2,1-0.2 s0.7,0.1,1,0.2c0.3,0.1,0.6,0.3,0.8,0.5c0.2,0.2,0.4,0.4,0.5,0.7c0.1,0.3,0.2,0.6,0.2,0.9s-0.1,0.6-0.2,0.9s-0.3,0.5-0.5,0.7 c-0.2,0.2-0.5,0.4-0.8,0.5c-0.3,0.1-0.6,0.2-1,0.2s-0.7-0.1-1-0.2s-0.5-0.3-0.8-0.5c-0.2-0.2-0.4-0.4-0.5-0.7S21.6,33.1,21.6,32.7z M25.8,28.1h-3.6L21.7,13h4.6L25.8,28.1z"
          fill="#fff"
        />
      </svg>
      <h1>
        404!
      </h1>
      <p>
        Sorry, it seems like you lost your way. Click this link and you'll be fine.
      </p>
      <a
        class="btn btn-primary"
        href="/login"
        role="button"
        tabindex="0"
      >
        Go to Login
      </a>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
